widgetry/src/widgets/button.rs

```diff
@@ -271,6 +308,507 @@ impl Btn {
+        self
+    }
+
+    /// Set a hotkey for the button
+    pub fn hotkey<MK: Into<MultiKey>>(mut self, key: MK) -> Self {
+        self.hotkey = Some(key.into());
+        self
+    }
+
+    /// Set a non-default tooltip [`Text`] to appear when hovering over the button.
+    ///
+    /// If a `tooltip` is not specified, a default tooltip will be applied.
+    pub fn tooltip(mut self, tooltip: Text) -> Self {
+        self.tooltip = Some(tooltip);
+        self
+    }
+
+    /// If a `tooltip` is not specified, a default tooltip will be applied. Use `no_tooltip` when
+    /// you do not want even the default tooltip to appear.
+    pub fn no_tooltip(mut self) -> Self {
+        // otherwise the widgets `name` is used
+        self.tooltip = Some(Text::new());
+        self
+    }
+
+    /// The button's items will be rendered in a vertical column
+    ///
+    /// If the button doesn't have both an image and label, this has no effect.
+    pub fn vertical(mut self) -> Self {
+        self.stack_axis = Some(geom_batch_stack::Axis::Vertical);
+        self
+    }
+
+    /// The button's items will be rendered in a horizontal row
+    ///
+    /// If the button doesn't have both an image and label, this has no effect.
+    pub fn horizontal(mut self) -> Self {
+        self.stack_axis = Some(geom_batch_stack::Axis::Horizontal);
+        self
+    }
+
+    /// The button cannot be clicked and will be styled as [`ControlState::Disabled`]
+    pub fn disabled(mut self) -> Self {
+        self.is_disabled = true;
+        self
+    }
+
+    /// Display the button's label before the button's image.
+    ///
+    /// If the button doesn't have both an image and label, this has no effect.
+    pub fn label_first(mut self) -> Self {
+        self.is_label_before_image = true;
+        self
+    }
+
+    /// Display the button's image before the button's label.
+    ///
+    /// If the button doesn't have both an image and label, this has no effect.
+    pub fn image_first(mut self) -> Self {
+        self.is_label_before_image = false;
+        self
+    }
+
+    /// Spacing between the image and text of a button.
+    /// Has no effect if the button is text-only or image-only.
+    pub fn stack_spacing(mut self, value: f64) -> Self {
+        self.stack_spacing = value;
+        self
+    }
+
+    // Building
+
+    /// Build a button.
+    ///
+    /// `action`: The event that will be fired when clicked
+    /// ```
+    /// # use widgetry::{Color, ButtonBuilder, ControlState, EventCtx};
+    ///
+    /// fn build_some_buttons(ctx: &EventCtx) {
+    ///     let one_off_builder = ButtonBuilder::new().label_text("foo").build(ctx, "foo");
+    ///
+    ///     // If you'd like to build a series of similar buttons, `clone` the builder first.
+    ///     let red_builder = ButtonBuilder::new()
+    ///         .bg_color(Color::RED, ControlState::Default)
+    ///         .bg_color(Color::RED.alpha(0.3), ControlState::Default)
```

Comment:
Should this second one be `::Hovering`?

Reply:
Whoops! Right it was intended to show different control state config. Fixed!